source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/MarketParticipants/Handlers/MarketParticipantDomainEventMapper.cs

```diff
@@ -29,6 +30,13 @@ public static class MarketParticipantDomainEventMapper
             var rolesUsedInChargesDomain = actorUpdatedIntegrationEvent.BusinessRoles
                 .Select(MarketParticipantRoleMapper.Map).ToList();
 
+            if (rolesUsedInChargesDomain.Count > 1)
+            {
+                throw new InvalidOperationException(
+                    $"Only 1 role per market participant with ID '{actorUpdatedIntegrationEvent.Gln}' is allowed, " +
+                    $"the current market participant has {rolesUsedInChargesDomain.Count} roles associated");
```

Comment:
nit: perhaps this will come in handy when coming across this error

```suggestion
                    $"Only 1 role per market participant with ID '{actorUpdatedIntegrationEvent.Gln}' is allowed, " +
                    $"the current market participant has {rolesUsedInChargesDomain.Count} roles associated in the integration event with id '{actorUpdatedIntegrationEvent.Id}'");
```